shared/storage.ts

```diff
@@ -312,34 +315,62 @@ class Storage {
     }
   }
 
+  private static async verifyCredential(
+    verifiableCredential: Credential,
+    issuer: string,
+  ) {
+    let isVerified = true;
+    if (isMosipVC(issuer)) {
+      const verificationResult = await verifyCredential(verifiableCredential);
+      isVerified = verificationResult.isVerified;
+    }
+    return isVerified;
+  }
+
   private static async loadVCs(completeBackupData: {}, encryptionKey: any) {
     try {
       const allVCs = completeBackupData['VC_Records'];
       const allVCKeys = Object.keys(allVCs);
       const dataFromDB = completeBackupData['dataFromDB'];
+
       // 0. Check for VC presense in the store
       // 1. store the VCs and the HMAC
-      allVCKeys.forEach(async key => {
-        let vc = allVCs[key];
-        const ts = Date.now();
-        const prevUnixTimeStamp = vc.vcMetadata.timestamp;
-        vc.vcMetadata.timestamp = ts;
-        dataFromDB.myVCs.forEach(myVcMetadata => {
-          if (
-            myVcMetadata.requestId === vc.vcMetadata.requestId &&
-            myVcMetadata.timestamp === prevUnixTimeStamp
-          ) {
-            myVcMetadata.timestamp = ts;
-          }
-        });
-        const updatedVcKey = new VCMetadata(vc.vcMetadata).getVcKey();
-        const encryptedVC = await encryptJson(
-          encryptionKey,
-          JSON.stringify(vc),
-        );
-        // Save the VC to disk
-        await this.setItem(updatedVcKey, encryptedVC, encryptionKey);
-      });
+      // 2. Verify the VC and update the result
+
+      await Promise.all(
+        allVCKeys.map(async key => {
+          let vc = allVCs[key];
```

Comment:
We can use const for vc as vc is not changed afterwards.

Reply:
Overlooked this, as did not change this part. Making the change.